### ISSUE TITLE
Why the training loss changed even the trainable-params is zero？

### ISSUE THREAD

**Issue by geekbutton** [keras-team/keras#8856]
I found that there is a shift for the  training loss, and when I tried to see why there is a shift, I set my own model to  non trainable:
![image](https://user-images.githubusercontent.com/13257051/34256650-4dd550c6-e691-11e7-8942-5bd249f5c1ed.png)
Then I  choose ten images as my training samples(they were also be used as the validation data), and set the batch-size to 5 and shuffle the data.Before I training my model, I do evaluate for my samples, the result is as below:
![image](https://user-images.githubusercontent.com/13257051/34256707-93529636-e691-11e7-8f7f-fa3b04684517.png)
Then I trained my model(actually it could not be trained)， and print the fit history, the result is as below:
![image](https://user-images.githubusercontent.com/13257051/34256735-bace98cc-e691-11e7-8b58-47f349909d8e.png)
It can be found that even the model is not be trained, and the validation loss never change, but the training loss were different among three epochs. And I found that if I set the shuffle to false, the result will be just right. I wanna know if this is a bug or the shuffle will make some difference when training.

**Comment by gabrieldemarmiesse** [comment]
Hi! 
I think this is because the training error is a running average of the error on each batch. This is noy the case with the validation loss, because it's a normal average. So the images appearing at the start of the training will have less impact on the training loss. That's why shuffling the data changes the training loss (I think). 
I hope it helps.

**Comment by geekbutton** [comment]
@gabrieldemarmiesse  Thank you very much.
I have read the source code about how the displayed loss is computed, I found that the displayed loss is a running average of the error on each batch just like you said. That can explain why the displayed loss changed each batch, but I am still confused why the final loss will be different while the model was not be trained.Shuffling the data only change the order of computation but the average loss should be the same while the model was not be trained.
And I found that if the batch-size is 1, then the result will be just right even shuffling the data.
